File: backend/app/schemas/store.py
import uuid
from typing import Optional
from sqlmodel import SQLModel

class StoreBase(SQLModel):
    name: str
    address: str
    description: Optional[str] = None
    logo_url: Optional[str] = None

class StoreCreate(StoreBase):
    latitude: Optional[float] = None
    longitude: Optional[float] = None

class StoreUpdate(SQLModel):
    name: Optional[str] = None
    address: Optional[str] = None
    description: Optional[str] = None
    logo_url: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None

class StorePublic(StoreBase):
    id: uuid.UUID
    owner_id: uuid.UUID
    latitude: Optional[float] = None
    longitude: Optional[float] = None

class StoreWithDistance(StorePublic):
    distance_km: Optional[float] = None

File: frontend/Wisebite/app/src/main/java/com/example/wisebite/data/model/Store.kt
package com.example.wisebite.data.model

import com.google.gson.annotations.SerializedName

data class Store(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("name")
    val name: String,
    
    @SerializedName("address")
    val address: String,
    
    @SerializedName("description")
    val description: String? = null,
    
    @SerializedName("logo_url")
    val logoUrl: String? = null,
    
    @SerializedName("owner_id")
    val ownerId: String? = null,
    
    @SerializedName("latitude")
    val latitude: Double? = null,
    
    @SerializedName("longitude")
    val longitude: Double? = null
) {
    // Helper methods
    val displayName: String
        get() = name
    
    val displayAddress: String
        get() = address
    
    val displayDescription: String
        get() = description ?: "Cửa hàng uy tín, chất lượng"
}